web/src/app/components/tasks-editor/tasks-editor.component.ts

```diff
@@ -0,0 +1,241 @@
+  [TaskType.PHOTO, TaskGroup.PHOTO],
+  [TaskType.DROP_PIN, TaskGroup.DROP_PIN],
+  [TaskType.DRAW_AREA, TaskGroup.DRAW_AREA],
+  [TaskType.CAPTURE_LOCATION, TaskGroup.CAPTURE_LOCATION],
+]);
+
+@Component({
+  selector: 'tasks-editor',
+  templateUrl: './tasks-editor.component.html',
+  styleUrls: ['./tasks-editor.component.scss'],
+})
+export class TasksEditorComponent {
+  formGroup!: FormGroup;
+
+  @Input() label?: string;
+  @Input() tasks?: List<Task>;
+  @Output() onChange?: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  addableTaskGroups: Array<TaskGroup> = [
+    TaskGroup.QUESTION,
+    TaskGroup.PHOTO,
+    TaskGroup.DROP_PIN,
+    TaskGroup.DRAW_AREA,
+    TaskGroup.CAPTURE_LOCATION,
+  ];
+
+  constructor(
+    private dataStoreService: DataStoreService,
+    private dialogService: DialogService
+  ) {}
+
+  ngOnChanges(): void {
+    const formBuilder = new FormBuilder();
+
+    if (this.tasks) {
+      this.formGroup = formBuilder.group({
+        tasks: formBuilder.array(
+          this.tasks
+            ?.toArray()
+            .slice(1)
+            .map(task =>
+              formBuilder.group({
+                id: task.id,
+                type: task.type,
+                required: task.required,
+                label: [task.label, Validators.required],
+                cardinality: task.multipleChoice?.cardinality,
+                options: formBuilder.array(
+                  task.multipleChoice?.options.toArray().map(option =>
+                    formBuilder.group({
+                      id: option.id,
+                      label: option.label,
+                      code: option.code,
+                    })
+                  ) || []
+                ),
+              })
+            ) || []
+        ),
+      });
+    } else {
+      const formBuilder = new FormBuilder();
+
+      this.formGroup = formBuilder.group({
+        tasks: formBuilder.array([]),
+      });
+    }
+
+    this.formGroup.statusChanges.subscribe(() => {
+      this.onChange?.emit(this.formGroup?.valid);
+    });
+  }
+
+  get formArray() {
+    return this.formGroup.get('tasks') as FormArray;
+  }
+
+  onTaskAdd(group: TaskGroup) {
+    const types = taskGroupToTypes.get(group);
+
+    const formGroup = new FormBuilder().group({
+      id: this.dataStoreService.generateId(),
+      type: types?.first(),
+      required: false,
+      label: ['', Validators.required],
+      cardinality: null,
+      options: new FormBuilder().array([]),
+    });
+
+    this.formArray.push(formGroup);
+  }
+
+  onDeleteTask(index: number) {
+    this.dialogService
+      .openConfirmationDialog(
+        'Warning',
+        'Are you sure you wish to delete this question? Any associated data ' +
+          'will be lost. This cannot be undone.'
+      )
+      .afterClosed()
+      .subscribe(dialogResult => {
+        if (dialogResult) {
+          this.formArray.removeAt(index);
+        }
+      });
+  }
+
+  onDuplicateTask(index: number) {
```

Comment:
I know this is still DRAFT - reminder to please add basic JSDoc comments to public methods.